### PR TITLE
[Fix] Update Text components to accept number values

### PR DESCRIPTION
**Pull request by ambrauer**
## Description / Motivation
Sitecore Headless Services was recently updated to return `number` instead of `string` type for number fields. This fix updates our SDKs to properly handle.

## Testing Details
Unit tests added. Manual testing for Next.js / React including editing in Experience Editor and Horizon.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

**Review by CobyPear — APPROVED** [review]

File: packages/sitecore-jss-vue/src/components/Text.test.ts [hunk 1]
```diff
@@ -75,6 +75,17 @@ describe('<Text />', () => {
     expect(rendered.element.innerHTML).toBe(props.field.value);
   });
 
+  it('should render number value', () => {
```
Comment (anastasiya29) [inline packages/sitecore-jss-vue/src/components/Text.test.ts:78]: Do we need to test negative numbers or zero value edge cases?

**Review by anastasiya29 — APPROVED** [review]

File: packages/sitecore-jss-react/src/components/Text.tsx [hunk 2]
```diff
@@ -90,10 +90,16 @@ export const Text: FunctionComponent<TextProps> = ({
 };
 
 Text.propTypes = {
-  field: PropTypes.shape({
-    value: PropTypes.string,
-    editable: PropTypes.string,
-  }),
+  field: PropTypes.oneOfType([
```
Comment (illiakovalenko) [inline packages/sitecore-jss-react/src/components/Text.tsx:93]: Instead of two separate shapes, you can write:
```
field: PropTypes.shape({
   value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
   editable: PropTypes.string
})
```

File: packages/sitecore-jss-vue/src/components/Text.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { CreateElement, FunctionalComponentOptions, RenderContext } from 'vue';
 export interface TextProps {
```
Comment (illiakovalenko) [inline packages/sitecore-jss-vue/src/components/Text.ts:3]: Let's add a unit test here? "should render number value"

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
See my comments below. Also you need to update values in styleguide.yml files, check this files:
https://github.com/Sitecore/jss/blob/bug/479677/samples/react/data/routes/styleguide/en.yml#L61
https://github.com/Sitecore/jss/blob/bug/479677/samples/nextjs/data/routes/styleguide/en.yml#L61

And add unit test for `sitecore-jss-angular`

File: packages/sitecore-jss-vue/src/components/Text.test.ts (hunk 1 above)
Comment (illiakovalenko) [reply packages/sitecore-jss-vue/src/components/Text.test.ts:78]: Definitely we need a test for `zero` value.

File: packages/sitecore-jss-react/src/components/Text.tsx (hunk 2 above)
Comment (ambrauer) [reply packages/sitecore-jss-react/src/components/Text.tsx]: Fixed

File: packages/sitecore-jss-vue/src/components/Text.ts (hunk 3 above)
Comment (ambrauer) [reply packages/sitecore-jss-vue/src/components/Text.ts:3]: Yep, we have tests for number value: https://github.com/Sitecore/jss/blob/bug/479677/packages/sitecore-jss-vue/src/components/Text.test.ts#L93

**Comment by ambrauer** [comment]
> See my comments below. Also you need to update values in styleguide.yml files, check this files:
> https://github.com/Sitecore/jss/blob/bug/479677/samples/react/data/routes/styleguide/en.yml#L61
> https://github.com/Sitecore/jss/blob/bug/479677/samples/nextjs/data/routes/styleguide/en.yml#L61
> 
> And add unit test for `sitecore-jss-angular`

@illiakovalenko Good catch with the yml files, done. We've also added tests for angular.

File: packages/sitecore-jss-vue/src/components/Text.ts (hunk 3 above)
Comment (illiakovalenko) [reply packages/sitecore-jss-vue/src/components/Text.ts:3]: Wow, sorry :D 
I missed

**Review by illiakovalenko — APPROVED** [review]
